### PR TITLE
remove max block height

### PR DESCRIPTION
**Pull request by mpatc**

### CODE REVIEW

**Comment by jgarzik** [comment]
`nHeight` arg no longer needed

**Comment by mpatc** [comment]
sry working on it

**Comment by kek-coin** [comment]
This seems to remove the 3 month upgrade period between segwit activation and the hardfork; is this intentional? If so, what is the rationale for removing this?

**Comment by opetruzel** [comment]
@kek-coin AFAICT, this only removes the Minimum_block_Height activation method that may have led to an activation on/around 21 September if it was longer.  The (+3 months) activation for the hardfork is still there (fSegwitSeasoned).

**Comment by mpatc** [comment]
as @opetruzel said, I think BIP102_FORK_BUFFER is redundant due to fSegwitSeasoned.

**Comment by sneurlax** [comment]
NACK

File: src/validation.cpp [hunk 1]
```diff
@@ -3072,7 +3072,7 @@ bool ContextualCheckBlock(const CBlock& block, CValidationState& state, const Co
             fHaveWitness = true;
         }
 
-        const CBlockIndex* pindexForkBuffer = pindexPrev->GetAncestor(nHeight - BIP102_FORK_BUFFER);
```
Comment (kek-coin) [inline src/validation.cpp]: This makes `fSegwitSeasoned` become true immediately upon Segwit reaching `THRESHOLD_ACTIVE`.

NACK

File: src/validation.cpp [hunk 2]
```diff
@@ -3072,7 +3072,7 @@ bool ContextualCheckBlock(const CBlock& block, CValidationState& state, const Co
...
+        const CBlockIndex* pindexForkBuffer = pindexPrev->GetAncestor(nHeight);
```
Comment (kek-coin) [inline src/validation.cpp]: This makes `fSegwitSeasoned` become true immediately upon Segwit reaching `THRESHOLD_ACTIVE`.

NACK

**Comment by kek-coin** [comment]
@mpatc @opetruzel `BIP102_FORK_BUFFER` is used to determine `fSegwitSeasoned`, removing it removes the 3 month buffer time. See my review.

**Comment by opetruzel** [comment]
@kek-coin 
You're right. Removing the Min Block Height logic is fine, but BIP102_FORK_BUFFER needs to stay in order for fSegwitSeasoned to work.

**Comment by jgarzik** [comment]
Superceded by #48

**Pull request closed without merging**